packages/ui/src/providers/runtime.provider.tsx

```diff
@@ -31,10 +32,18 @@ export const RuntimeProvider: FunctionComponent<PropsWithChildren<{ catalogUrl:
   }
 
   const [selectedCatalog, setSelectedCatalog] = useState<CatalogLibraryEntry | undefined>(localSelectedCatalog);
-  const basePath = props.catalogUrl.substring(0, props.catalogUrl.lastIndexOf('/'));
+  const settingsAdapter = new LocalStorageSettingsAdapter();
+  const settingsCatalogUrl = settingsAdapter.getSettings().catalogUrl;
+  let catalogUrl;
+  if (isDefined(settingsCatalogUrl) && settingsCatalogUrl !== '') {
+    catalogUrl = settingsCatalogUrl;
+  } else {
+    catalogUrl = `${CatalogSchemaLoader.DEFAULT_CATALOG_PATH}`;
+  }
+  const basePath = catalogUrl.substring(0, catalogUrl.lastIndexOf('/'));
 
   useEffect(() => {
-    fetch(props.catalogUrl)
+    fetch(catalogUrl)
```

Comment:
I think we can restore this file, as we're passing the `catalogUrl` through the props of the react component.

For VSCode, we're passing it from the `KaotoEditorFactory` as a `constructor` argument for the `KaotoEditorApp` component.

For the online version, we do this in the `App` component

Reply:
effectively it is working, I guess I did not recompiled everything correctly when i tried before